### DISCUSSION TITLE
Set the Legend Properties

### DISCUSSION THREAD

**Discussion by RoyiAvital** [cjdoris/Bokeh.jl#29] [Q&A]
Is there a way, beside `hF.legend.click_policy = "hide";` to set the properties of the legend of the a figure?

I tried:

```julia
hL = Legend(;click_policy = "hide");
hF = figure(;width = 900, height = 900, title = "Center of Mass", 
			x_axis_label = "x", y_axis_label = "y", tooltips = [("x", "\$x"), ("y", "\$y")]);
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "1");
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "2");
plot!(hF, hL);
```

Which didn't work.
I also tried:

```julia
hF = figure(;width = 900, height = 900, title = "Center of Mass", 
			x_axis_label = "x", y_axis_label = "y", tooltips = [("x", "\$x"), ("y", "\$y")]);
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "1");
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "2");
plot!(hF, Legend; click_policy = "hide");
```

Which didn't work either.

I wonder, how should the function `Legend` be used?

**Comment by cjdoris** [comment]
Either of the other ways you tried work fine if you `plot!` the legend before the scatters, e.g.
```julia
hF = figure(;width = 900, height = 900, title = "Center of Mass", 
			x_axis_label = "x", y_axis_label = "y", tooltips = [("x", "\$x"), ("y", "\$y")]);
plot!(hF, Legend; click_policy = "hide");
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "1");
plot!(hF, Scatter; x = rand(5), y = rand(5), legend_label = "2");
```

This is because when you plot a scatter (or any glyph) and use one of the `legend_*` arguments, then it either uses the pre-existing legend or creates a new legend if there isn't already one on the figure. So the way you were doing it, the first `plot!(hF, Scatter; ...)` created a new legend containing the two labels, and your `plot!(hf, Legend; ...)` added a second blank legend. Whereas putting `plot!(hF, Legend; ...)` first creates the legend with the properties you want, and then that gets added to by the subsequent `plot!(hF, Scatter; ...)` calls.

**Reply by RoyiAvital** [reply]
@cjdoris , Makes sense!  

Thank You.
